src/rgw/driver/sfs/types.cc

```diff
@@ -149,6 +149,12 @@ Object* Object::try_fetch_from_database(
     return nullptr;
   }
 
+  // don't return the version_id if versioning is not enabled
+  // and the user didn't ask for a specific version.
+  // Returning the version_id when no required breaks s3tests related to tags
```

Comment:
I think it would be more important if we stated that it breaks s3 semantics, rather than the s3tests themselves.

Also, s/when no/when not/